api/src/routes/donate.ts

```diff
@@ -214,3 +201,164 @@ export const donateRoutes: FastifyPluginCallbackTypebox = (
 
   done();
 };
+
+/**
+ * Plugin for the donation endpoints.
+ *
+ * @param fastify The Fastify instance.
+ * @param _options Options passed to the plugin via `fastify.register(plugin, options)`.
+ * @param done The callback to signal that the plugin is ready.
+ */
+export const chargeStripeRoute: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  // Stripe plugin
+  const stripe = new Stripe(STRIPE_SECRET_KEY, {
+    apiVersion: '2024-06-20',
+    typescript: true
+  });
+
+  // @ts-expect-error - @fastify/csrf-protection needs to update their types
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  fastify.addHook('onRequest', fastify.csrfProtection);
+
+  fastify.post(
+    '/donate/create-stripe-payment-intent',
+    {
+      schema: schemas.createStripePaymentIntent
+    },
+    async (req, reply) => {
+      const { email, name, amount, duration } = req.body;
+
+      if (
+        !isEmail(email) ||
```

Comment:
Nit: we can use `Type.String({ format: 'email', maxLength: 1024 })` in the validator. Not a big deal, though.

Reply:
you mean the schema? I updated that accordingly.

Reply:
Cool, I'll add a suggestion for how this can be cleaned up.